Packs/CybleEventsV2/Integrations/CybleEventsV2/CybleEventsV2.py

```diff
@@ -94,15 +288,213 @@ def test_response(client, method, base_url, token):
+        input_params['end_date'] = args.get('end_date', '')
+
+        if not args.get('end_date', ''):
+            input_params['end_date'] = datetime.utcnow().astimezone().isoformat()
+
+    else:
+        initial_interval = demisto.params().get('first_fetch_timestamp', 1)
+
+        if 'event_pull_start_date' not in last_run.keys():
+            event_pull_start_date = datetime.utcnow() - timedelta(days=int(initial_interval))
+            input_params['start_date'] = event_pull_start_date.astimezone().isoformat()
+
+        else:
+            input_params['start_date'] = last_run['event_pull_start_date']
+
+        input_params['end_date'] = datetime.utcnow().astimezone().isoformat()
+
+    latest_created_time = input_params['start_date']
+
+    final_input_structure = alert_input_structure(input_params)
+
+    all_alerts = set_request(client, method, token, final_input_structure, url)
+    timestamp_count = {}
+
+    for alert in all_alerts:
+
+        timestamp = alert['created_at']
+
+        if timestamp in timestamp_count.keys():
+            timestamp_count[timestamp] += 1
+        else:
+            timestamp_count[timestamp] = 1
+
+    alert_count = 0
+    prev_timestamp = all_alerts[0].get('created_at')
+    last_timestamp = all_alerts[-1].get('created_at')
+
+    alerts = []
+
+    for alert in all_alerts:
+
+        current_timestamp = alert.get('created_at')
+
+        if current_timestamp == prev_timestamp:
+            alerts.append(alert)
+        else:
+            alert_count += timestamp_count[prev_timestamp]
+            prev_timestamp = current_timestamp
+
+            if alert_count + timestamp_count[current_timestamp] <= max_fetch and current_timestamp != last_timestamp:
+                alerts.append(alert)
+            else:
+                break
+
+    del all_alerts
+    del timestamp_count
+
+    incidents = []
+
+    if alerts:
+
+        timestamp_str = alerts[-1].get('created_at')
+        original_datetime = datetime.strptime(timestamp_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+        updated_datetime = original_datetime + timedelta(microseconds=1000)
+        latest_created_time = updated_datetime.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+        events = format_incidents(alerts, hide_cvv_expiry)
+
+        for event in events:
+
+            inci = {
+                'name': event.get('name'),
+                'severity': event.get('severity'),
+                'rawJSON': json.dumps(event),
+                'alert_group_id': event.get('alert_group_id'),
+                'event_id': event.get('event_id'),
+                'keyword': event.get('keyword'),
+                'created': event.get('created_at')
+            }
+            incidents.append(inci)
+
+        next_run = {'event_pull_start_date': latest_created_time}
+
+        return incidents, next_run
+    else:
+        return [], {'event_pull_start_date': latest_created_time}
+
+
+def update_remote_system(client, method, token, args, url):
+
```

Comment:
Please add a doc string to the new functions you added, explaining what the functions do, what does it expect as input and output.

Reply:
Done